modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala

```diff
@@ -125,6 +126,16 @@ class StateChannel[F[_]: Async: KryoSerializer: SecurityProvider: Random](
       storages.block.store(fb.hashedBlock).handleErrorWith(e => logger.debug(e)("Block storing failed."))
     }
 
+  private val sendBlockToL0: Pipe[F, FinalBlock, Unit] =
+    _.evalMap { fb =>
+      (for {
+        tips <- storages.block
+          .pullTips(appConfig.tips.maximumTipsCount)
+        // l0Peer <- storages.l0Cluster.getPeers.map(Random.shuffle).map(_.head)
+        // req <- p2PClient.l0DAGCluster.sendL1Output(L1Output(fb.hashedBlock.signed, tips))(l0Peer)
+      } yield ()).handleErrorWith(e => logger.warn(e)("Sending block to L0 failed."))
```

Comment:
if request fails, you made `successful` so it will not throw an error but return a boolean whether succeeded or no. check out my second comment, I'd recommend to handle boolean and `handleErrorWith` will catch the error from the flow, not from the API